src/drivers/QtTestDriver.cpp

```diff
@@ -8,25 +8,32 @@ namespace cucumber {
 namespace internal {
 
 const InvokeResult QtTestStep::invokeStepBody() {
-    QTemporaryFile file;
-    if (!file.open()) {
-        return InvokeResult::failure("Unable to open temporary file needed for this test");
+    QString file_name;
+    {
+        QTemporaryFile file;
+        if (!file.open()) {
+            return InvokeResult::failure("Unable to open temporary file needed for this test");
+        }
+        file.close();
     }
-    file.close();
 
-    QtTestObject testObject(this);
+    file_name += ".txt";
```

Comment:
The file name is always ".txt" as it is not derived from the temporary file (I'd expect this was the intention).

Reply:
right, should be file_name = file.fileName() + ".txt"